lib/perl/Genome/InstrumentData/AlignmentResult/Merged.pm

```diff
@@ -219,24 +226,70 @@ sub create {
         for my $alignment (@alignments) {
             $alignment->add_user(user => $self, label => 'uses');
         }
-
-        return 1;
-    };
-    if (my $error = $@) {
-        $tx->rollback();
-        die $error;
-    } elsif ($rv ne 1) {
+    }
+    catch {
         $tx->rollback();
-        $self->error_message('Unexpected return value: ' . $rv);
-        die $self->error_message;
-    } else {
-        $tx->commit();
+        die $self->error_message('Merge failed due to error: ' . $_);
     }
+    finally {
+        for my $allocation (@temp_allocations) {
+            $allocation->delete;
+        }
+        unless ($_) {
+            $tx->commit();
```

Comment:
Does this mean the transaction won't be committed if the allocation deletes fail?  Is that what we want?  Or would it make sense to move the commit to the end of the try block?